upcloud/resource_upcloud_server_test.go

```diff
@@ -954,3 +957,142 @@ func TestUpcloudServer_createPreChecks(t *testing.T) {
 		},
 	})
 }
+
+func configHotResize(planName string, hotResize bool, captureUptime bool, checkUptime bool, keyDir string) string {
+	provisioner := ""
+
+	if captureUptime {
+		provisioner = `
+			provisioner "remote-exec" {
+				inline = [
+					"uptime -s > /tmp/server_start_time.txt",
+				]
+				connection {
+					type        = "ssh"
+					user        = "root"
+					host        = self.network_interface[0].ip_address
+					private_key = file("%s/id_rsa")
+				}
+			}
+		`
+		provisioner = fmt.Sprintf(provisioner, keyDir)
+	} else if checkUptime {
+		provisioner = `
+			provisioner "remote-exec" {
+				inline = [
+					"cat /tmp/server_start_time.txt > /tmp/original_start_time.txt",
+					"uptime > /tmp/current_start_time.txt",
+					"cmp /tmp/original_start_time.txt /tmp/current_start_time.txt && echo 'PASS: Server did not restart' || echo 'FAIL: Server restarted'",
+				]
+				connection {
+					type        = "ssh"
+					user        = "root"
+					host        = self.network_interface[0].ip_address
+					private_key = file("%s/id_rsa")
+				}
+			}
+		`
+		provisioner = fmt.Sprintf(provisioner, keyDir)
+	}
+
+	return fmt.Sprintf(`
+		resource "upcloud_server" "hot_resize" {
+			hostname    = "tf-acc-test-server-hot-resize"
+			zone        = "pl-waw1"
+			plan        = "%s"
+			metadata    = true
+			hot_resize  = %t
+
+			login {
+				user = "root"
+				keys = [
+					file("%s/id_rsa.pub")
+				]
+			}
+
+			template {
+				storage = "%s"
+				size    = 10
+			}
+
+			network_interface {
+				type = "public"
+			}
+
+			%s
+		}
+
+		output "server_ip" {
+			value = upcloud_server.hot_resize.network_interface[0].ip_address
+		}
+	`, planName, hotResize, keyDir, debianTemplateUUID, provisioner)
+}
+
+// generateSSHKey generates an SSH key pair in the given directory
+func generateSSHKey(t *testing.T, keyDir string) error {
```

Comment:
Could also use [tls_private_key](https://registry.terraform.io/providers/hashicorp/tls/latest/docs/resources/private_key) resource to generate the key 🤔 Both approaches should work very similarly, though.

Reply:
I think the downside of `tls_private_key` is that it's not a core Terraform resource. So it requires extra provider dependencies that will have to be initialised in `init()` alongside of the UpCloud provider. This requires a new Go dependency. Maybe opting in to the openssh CLI interface just might be better?

Reply:
Actually maybe a pure Go implementation for key generation would be better. Maybe someone wants to run acceptance tests on Windows. :)

Reply:
Done in https://github.com/UpCloudLtd/terraform-provider-upcloud/pull/747/commits/330cf5df839d7f31be77d33233661db76b9bb878